core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -155,6 +181,12 @@ public double getUtilization()
         return bufferedBytes.get() / (double) maxBufferedBytes;
     }
 
+    public synchronized TDigestHistogram getUtilizationHistogram()
+    {
+        recordBufferUtilization();
+        return new TDigestHistogram(TDigest.copyOf(bufferUtilization));
```

Comment:
Just return copy of `TDigest` here.

Did you figure if `copyOf` can be expensive?

Reply:
Yes, `copyOf` is in some sense expensive. I see that `Total Blocked Time` increases from ~6s to ~2min on `OutputBufferMemoryManager`'s monitor.

Reply:
No, it is not expensive. No visibile effect in serial benchmarks (CPU time) and not visible effects in concurrency benchmakrs (I mean no higher lock contention on `OutputBufferMemoryManager`)